### ISSUE TITLE
Document min releases requirement for tools

### ISSUE THREAD

**Issue by IgorTodorovskiIBM** [ZOSOpenTools/meta#561]
We may need to consider specific tools and their requirements as well.

Also, Go has a soft requirement for V2R5.

Perhaps this can be a buildenv setting which gets propagated to the zopen_releases.json / metadata.json

**Comment by MikeFultonDev** [comment]
yes - it makes sense to have that in your metadata info - just another 'runtime pre-req'... Will let you use the range (e.g. z/OS 2.4 and up for most and z/OS 2.5 for go... maybe z/OS 3.1 for something cool and new?)

**Comment by IgorTodorovskiIBM** [comment]
We may also need something a bit more granular. Brew has "depends_on" - https://github.com/search?q=org%3AHomebrew+maximum_macos&type=code

procps has many bulitin assumptions on the /proc filesystem being present. 

We could add a new var in the buildenv as such:

```
ZOPEN_RUNTIME_OS_DEPENDENCIES="procfs os>=3.1"
```
We could start with a hardcoded list. procfs will check for the presence of the /proc file system. If needed, we could also add custom checks.

zopen build would propagate this list to the metadata.json. Then on `zopen install`, it would use the logic to check if the dependencies are satisfied and if not, exit with an error.

**Comment by IgorTodorovskiIBM** [comment]
Related PR: https://github.com/ZOSOpenTools/meta/pull/844
